Source/dvlnet/tcp_client.cpp

```diff
@@ -13,25 +13,56 @@
 #include "options.h"
 #include "utils/language.h"
 #include "utils/str_cat.hpp"
+#include "utils/str_split.hpp"
 
 namespace devilution::net {
 
-int tcp_client::create(std::string addrstr)
+int tcp_client::create(std::string_view addrstr)
 {
 	auto port = *sgOptions.Network.port;
-	local_server = std::make_unique<tcp_server>(ioc, addrstr, port, *pktfty);
+	local_server = std::make_unique<tcp_server>(ioc, std::string(addrstr), port, *pktfty);
 	return join(local_server->LocalhostSelf());
 }
 
-int tcp_client::join(std::string addrstr)
+int tcp_client::join(std::string_view addrstr)
 {
 	constexpr int MsSleep = 10;
 	constexpr int NoSleep = 250;
 
-	std::string port = StrCat(*sgOptions.Network.port);
+	const char *defaultPort = "6112";

```

Comment:
Perhaps we want to use `*sgOptions.Network.port` as the default? Don't know if it makes sense, but currently the setting still exists and is saved to the ini file, but will do nothing after this PR.

Reply:
It's still used as the TCP server port for hosting games.

https://github.com/diasurgical/devilutionX/blob/73720267adcb55d8e4912638ab24ca1d6046fa82/Source/dvlnet/tcp_client.cpp#L22-L23

I chose to hardcode the default for joining games because I figured it would be less confusing. This way, you change the server port if you have a port conflict for hosting, not if you have a friend that has a port conflict.